source/Engine.cpp

```diff
@@ -1257,7 +1257,7 @@ void Engine::EnterSystem()
 			}
 		};
 		for(const auto &hazard : system->Hazards())
-			CreateWeather(hazard, Point());
+			CreateWeather(hazard, hazard.Get()->SystemWide() ? flagship->Position() : Point());
```

Comment:
This change makes it so that hazards with a force (like the black hole hazards) are centered around the flagship for system-wide hazards, which is really weird.

I don't really see a reason why system-wide hazards should have a different center than ranged ones anyways.

Reply:
While I didn't intend this when I made the request, this could actually be useful for creating some interesting gameplay, for instance, having a system that has some sort of hazard that projects an effect onto the target ship. Although perhaps whether or not the center is moved to the player's ship should be controlled by a specified attribute, rather than simply assuming that all system-wide stuff is centered on the player.

If I recall the explanation correctly, this is changed so that visual effects can be displayed in an area more or less restricted to the player's field of view. Otherwise, the hazard needs to try to carpet everything out to infinity with visual effects. This is actually the real reason why the EW hazards have a limit at all, and  have one as low they do. Having enough effects being rendered to make a decent looking density when spread out over a 100000 radius area had severe performance impacts.

Reply:
Good point, and looking at the code again I was wrong this doesn't actually change the centering, oops.